### PR TITLE
Improve contextual guidance on small screens

### PR DESCRIPTION
**Pull request by alex-ju**
This PR updates the contextual guidance component and document edit page after a design iteration to improve the user interface on small screens or when using magnifying tools.

Changes:
- Set highlight border on the left for mobile views b13d318
- Increase margin-bottom for the guidance container for mobile views cbf4e43
- Hide the input length suggester container when no message is presented d03ee7a
- Add documentation to contextual guidance and update the example 534029b
- Update edit page to place the url preview section in a new row 51b1050
- Remove inset text top margin when used in the url preview component f295751

## Review app preview
[Preview contextual guidance component](https://content-publisher-revi-pr-1253.herokuapp.com/component-guide/contextual_guidance)
[Preview the component in edit document page](https://content-publisher-revi-pr-1253.herokuapp.com/documents/4571577e-63f9-4879-b36b-3b257b33be24:en/edit)

### Title field
contextual-guidance-title

### Summary field
imgonline-com-ua-twotoone-0QOopfX0Jk9Q

[Trello card](https://trello.com/c/OHO5MB3o)

### CODE REVIEW

File: app/views/components/docs/contextual_guidance.yml [hunk 1]
```diff
@@ -1,5 +1,16 @@
 name: Contextual guidance
 description: Provides a container with additional information when focusing an input field (e.g. input, textarea)
+body: |
+  This component is build to be used with a focusable elements that refers to it.
+
+  The contextual guidance must have an id. The input must refer the contextual guidance using a `data-contextual-guidance`
+  attribute (e.g. `<input data-contextual-guidance="my-contextual-guidance-id">`).
+
+  The guidance container is set to float on the page and prevent other elements from moving around (e.g. pushing next fields down the page).
```
Comment (benthorner) [inline app/views/components/docs/contextual_guidance.yml]: Might be worth clarifying this applies to desktop mode.

File: app/assets/stylesheets/components/_url-preview.scss [hunk 2]
```diff
@@ -2,6 +2,10 @@
   @include govuk-font(19);
 
   @include govuk-responsive-margin(6, "bottom");
+
+  .govuk-inset-text {
+    margin-top: 0;
```
Comment (benthorner) [inline app/assets/stylesheets/components/_url-preview.scss]: This one confused me a bit - as discussed, it only applies in narrow width views.

File: app/views/components/docs/contextual_guidance.yml (hunk 1 above)
Comment (alex-ju) [reply app/views/components/docs/contextual_guidance.yml]: Good all. Updated to clarify that applies to mobile and desktop only.

File: app/assets/stylesheets/components/_url-preview.scss (hunk 2 above)
Comment (alex-ju) [reply app/assets/stylesheets/components/_url-preview.scss]: After the url preview component markup on the edit page has been move on a new row the margin-top of the inset text element creates unnecessary whitespace between it and the title field above (left). We do not want this as the page address should be visually associated with the title so we're reducing the spacing to match the original (right).

I updated the commit description too trying to explain the need for it.

url-preview-margin

**Pull request merged**